gwpy/timeseries/tests/test_timeseries.py

```diff
@@ -1052,6 +1052,13 @@ def test_q_transform_logf(self, losc):
         assert qspecgram.q == 5.65685424949238
         nptest.assert_almost_equal(qspecgram.value.max(), 155.93774, decimal=5)
 
+    def test_q_transform_nan(self, losc):
+        data = TimeSeries([numpy.nan for y in losc.value])
+        data.__array_finalize__(losc)
```

Comment:
I would just create a new `TimeSeries` of a fixed size for this test, rather than pulling in any real data

```python
        data = TimeSeries(numpy.empty(<N>) * numpy.nan, ...)
```

Reply:
@duncanmmacleod, good call, this is done in the latest commit.